libs/ha/ha_common_libs_k8s.py

```diff
@@ -1308,3 +1312,48 @@ def simulate_disk_cvg_failure(node_obj, source, resource_status, resource_type,
             return False, error
 
         return True, config_dict
+
+    @staticmethod
+    def get_node_resource_ids(node_obj, r_type, n_type=None, node_id=None):
+        """
+        :param node_obj: Object of master node
+        :param r_type: Type of resource (node, disk, cvg)
+        :param n_type: Type of the node (data, server)
+        :param node_id: ID of node from which resource IDs to be extracted (Required only for
+        disk and cvg)
+        :return: List
+        """
+        switcher = {
+            'node': {
+                'data': {
+                    'cmd': common_cmd.GET_DATA_NODE_ID_CMD.format(
+                        common_const.MOCK_MONITOR_REMOTE_PATH)},
+                'server': {
+                    'cmd': common_cmd.GET_SERVER_NODE_ID_CMD.format(
+                        common_const.MOCK_MONITOR_REMOTE_PATH)}},
+            'disk': {
+                'cmd': common_cmd.GET_DISK_ID_CMD.format(common_const.MOCK_MONITOR_REMOTE_PATH,
+                                                         node_id)},
+            'cvg': {
+                'cmd': common_cmd.GET_CVG_ID_CMD.format(common_const.MOCK_MONITOR_REMOTE_PATH,
+                                                        node_id)}
+        }
+
+        LOGGER.info("Get HA pod name for publishing event")
+        ha_pod = node_obj.get_all_pods(pod_prefix=common_const.HA_POD_NAME_PREFIX)[0]
+        if r_type == 'node':
+            cmd = switcher[r_type][n_type]['cmd']
+        else:
+            cmd = switcher[r_type]['cmd']
+        LOGGER.info("Running command %s on pod %s", cmd, ha_pod)
+        try:
+            resp = node_obj.send_k8s_cmd(operation="exec", pod=ha_pod,
+                                         namespace=common_const.NAMESPACE + " -- ",
+                                         command_suffix=cmd, decode=True)
+            resp = literal_eval(resp)
+        except IOError as error:
+            LOGGER.error("Failed to get resource IDs for %s", r_type)
+            raise error
```

Comment:
Why raise error instead of returning  ?

Reply:
Don't want to return bool from here, as it will add some more actions in calling function. And if bool False is returned, we are anyway going to return an error from calling function. So raising error here only.

Reply:
@ApurwaMohite  I don't see point in handling even the IOError . Let this function propagate even that.  Alternatively if you intend to log the exception then log the complete traceback instead of error.